src/atmospheric_thermodynamics.jl

```diff
@@ -28,18 +28,21 @@ struct Saturation{FT}
     triple_point_pressure :: FT
 end
 
-Adapt.adapt_structure(to, sat::Saturation) =
-    Saturation(adapt(to, sat.energy_reference_temperature),
-               adapt(to, sat.triple_point_temperature),
-               adapt(to, sat.triple_point_pressure))
+function Adapt.adapt_structure(to, sat::Saturation)
+    T₁ = adapt(to, sat.energy_reference_temperature)
+    T₂ = adapt(to, sat.triple_point_temperature)
+    p₃ = adapt(to, sat.triple_point_pressure)
+    FT = typeof(T₁)  # assume all same type
+    return Saturation{FT}(T₁, T₂, p₃)
```

Comment:
```suggestion
    return Saturation(T₁, T₂, p₃)
```

Reply:
You don't need the explicit `FT` here because it can be inferred; possibly also better design since it will fail rather than convert to `FT`

Reply:
you sure about this?

Reply:
for example,

```julia
julia> struct Test{T}
       a :: T
       b :: T
       end

julia> t = Test(1, 2)
Test{Int64}(1, 2)

julia> t = Test(1.0, 2.0)
Test{Float64}(1.0, 2.0)
```

Reply:
unless I am confused about what you are trying to accomplish?

Reply:
Specifying the type parameter has the effect of calling `convert` on all arguments that require that type. So if we _want_ to call convert, we should write the code this way. However, in this case we are constructing something via `adapt`, so we should not need to call convert. therefore the extra call to convert is, in my opinion, confusing, because it implies there is a need to call convert when there is not. For example:

```julia
julia> t = Test(1, 2.0)
ERROR: MethodError: no method matching Test(::Int64, ::Float64)

Closest candidates are:
  Test(::T, ::T) where T
   @ Main REPL[1]:2

Stacktrace:
 [1] top-level scope
   @ REPL[4]:1

julia> t = Test{Float64}(1, 2.0)
Test{Float64}(1.0, 2.0)
```

Reply:
thanks for the explanation